src/services/validation.service.js

```diff
@@ -1,93 +1,112 @@
+    },
+    min: function (value, $scope, attr) {
+        return parseFloat(value) >= parseFloat(attr.min);
+    },
+    max: function (value, $scope, attr) {
+        return parseFloat(value) <= parseFloat(attr.max);
+    },
+    length: function (value, $scope, attr) {
+        return value.length === parseInt(attr.length);
+    },
+    strictemail: function(value, $scope, attr) {
+      return  new RegExp(/^[_a-z0-9]+(\.[_a-z0-9]+)*@[a-z0-9-]+(\.[a-z0-9-]+)*(\.[a-z]{2,4})$/).test(value);
+    }
+  };
 
-    var selectors = [];
-    var elements = ["input", "select", "textarea"];
+  var selectors = [];
+  var elements = ['input', 'select', 'textarea'];
 
-    angular.forEach(elements, function (element) {
-        selectors.push(element + "[ng-model]");
-        selectors.push(element + "[data-ng-model]");
-    });
+  angular.forEach(elements, function (element) {
+    selectors.push(element + '[ng-model]');
+    selectors.push(element + '[data-ng-model]');
+  });
 
-    var selector = selectors.join(",");
+  var selector = selectors.join(',');
 
-    return {
-        /**
-         * Search all the input element inside the given DOM element and apply the "bs-validation" directive so we
-         * need not a add it for every form element.
-         * @param $element
-         */
-        addDirective: function ($element) {
-            var validateableElements = $element.findAll(selector);
-            validateableElements.attr("bs-validation", "");
-            return validateableElements;
-        },
-        addToNgIncludedURLs: function (url) {
-            if (ngIncludedURLs.indexOf(url) === -1) {
-                ngIncludedURLs.push(url);
-            }
-        },
-        addValidator: function ($scope, $attr, ngModelController, validatorKey) {
-            ngModelController.$validators[validatorKey] = function (modelValue, viewValue) {
-                var value = modelValue || viewValue;
-                return ngModelController.$isEmpty(value) || genericValidators[validatorKey](value, $scope, $attr);
-            };
-        },
-        checkNgIncludedURL: function (url) {
-            var index = ngIncludedURLs.indexOf(url);
-            if (index > -1) {
-                ngIncludedURLs.splice(index, 1);
-                return true;
-            }
+  return {
+    /**
+     * Search all the input element inside the given DOM element and apply the 'bs-validation' directive so we
+     * need not a add it for every form element.
+     * @param $element
+     */
+    getValidators: function (){
+      var builtIn = ['equalto', 'min', 'max', 'number', 'digits', 'length']
+      var additional = Object.keys(genericValidators);
+      return builtIn.concat(additional);
+    },
 
-            return false;
-        },
-        getDefaultMessage: function (key) {
-            return messages[key];
-        }
-    };
-});
\ No newline at end of file
+    getMeta: function (){
+      return ['matchName'];
```

Comment:
actually if you get a 422 from the Server with some meta information like an error message for each field.
You can set the validity of each form element with:
```javascript
let f = this.$scope.form;
    [ f.email, f.password, f.password_confirmation].forEach(
      item => item.$setValidity('response', false)
    );
```
and show the error with: ` response-notification="{{su.error.email}}" ` 

thats awesome.
but you have to validate the field 'on-blur' manually in the controller. But thats acceptable